### ISSUE TITLE
UnifiedDiffParserException: expected file start line not found

### ISSUE THREAD

**Issue by elanzini** [java-diff-utils/java-diff-utils#122]
**Describe the bug**
I tried to parse [this diff](http://hg.code.sf.net/p/graphicsmagick/code/raw-diff/135bdcb88b8d/coders/wpg.c) but I am getting `Exception in thread "main" com.github.difflib.unifieddiff.UnifiedDiffParserException: expected file start line not found`.

**To Reproduce**
You can simply save the diff as `diff.txt` file and then parse it as follows:
``` java
FileInputStream stream = new FileInputStream(new File("/path/to/diff.txt"));
UnifiedDiff patch = UnifiedDiffReader.parseUnifiedDiff(stream);                      <-- ERROR
```

**Expected behavior**
I expected the `UnifiedDiffReader` to be able to parse the provided diff.

**Comment by elanzini** [comment]
As far as I can tell from some debugging, it has to do with the empty lines at the end of the diff.
If you take the patch and delete the empty lines, it is able to parse it.

**Comment by wumpz** [comment]
The latest snapshot version resolves this issue.

**Comment by challarao** [comment]
I'm not sure if I should open a new issue/question. But I face this issue from my unit tests.

"expected data line not found"

I was able to track it down empty lines in place of neutral lines. For example:

`
diff --git a/v1 b/v1
--- a/v1
+++ b/v1

+asd
+
 some=code
 else=code
<here there is an empty line instead of a neutral line with space>
+add
+add
`


This happened because Intellij keeps trimming spaces to empty strings. As I maintain these diffs in text files in my repo for unit tests, when I open them I noticed that they are being trimmed. Glad I could found the issue. Otherwise, I wonder how many hours I would have spent as it's difficult track it down.

While this is not an issue in the parser, I would just like to take an opinion/report to you whether this should be tolerated by the UnifiedReader?
